core/src/main/java/org/bitcoinj/wallet/CoinSelection.java

```diff
@@ -33,8 +34,18 @@ public class CoinSelection {
     public final Coin valueGathered;
     public final Collection<TransactionOutput> gathered;
 
+    public CoinSelection(Collection<TransactionOutput> gathered) {
+        this(sumOutputValues(gathered), gathered);
+    }
+
     public CoinSelection(Coin valueGathered, Collection<TransactionOutput> gathered) {
         this.valueGathered = valueGathered;
         this.gathered = gathered;
     }
+
+    private static Coin sumOutputValues(Collection<TransactionOutput> outputs) {
```

Comment:
I've also thought adding a `static` `sum()` method to `TransactionOutput` (and `TranactionInput`) might be useful. We're summing them up in many places.

Reply:
I'll look for them, but I think unconditional sums are not that often.

Reply:
There's at least a handful. I've done this more than once, but I don't think it ever made it into a pull request.

Reply:
Actually here's a PR: https://github.com/bitcoinj/bitcoinj/pull/2412

I actually realized that it's better to use `reduce` and `Coin::add` after making that PR. There's also the issue with nullable in the values for `TransactionInput`